### PR TITLE
Prepare model training examples

### PR DESCRIPTION
**Pull request by Korred**
- added simple examples on how to train a model and predicts image masks
- added function to covert a softmax-ed ndarray mask to argmax
- added the dice coefficient loss function (based on https://github.com/maju116/pyplatypus/blob/7b1706e4d6f2ef184d0e90d87226506cebb8ed1d/pyplatypus/segmentation/loss_functions.py#L91)
- added a train/test/validation split function for a list of ImageMaskPairPaths
- ensured the utility folder can be used as a module

### CODE REVIEW

**Review by Katarzyna67 — APPROVED** [review]
ok :)
